### DISCUSSION TITLE
FidelityFX Upscaling in Upscaling Filter

### DISCUSSION THREAD

**Discussion by BuyMyMojo** [Xaymar/obs-stream-effects#851] [Ideas & Feedback]
### Explain the Feature/Enhancement

The option to use AMDs new FSR for source upscale, a more lightweight upscaler that isn't gpu brand dependant and a decent enough scaler

https://github.com/GPUOpen-Effects/FidelityFX-FSR

### What makes this feature/enhancement useful for StreamFX and its users?

A new option for the upscale filter that is light weight and more useful for single PC streaming

### Checklist

- [ ] I am offering a monetary bounty/reward for the implementation of this Feature/Enhancement.
- [X] There is no other way to currently do this in StreamFX or OBS Studio.

### Any additional Information we need to know?

Could supply just a shader file to do it too? https://www.shadertoy.com/view/stXSWB

also here is another software that has it implemented to scale any window to be full screen: https://github.com/Blinue/Magpie

**Comment by Xaymar** [comment]
FSR is an Edge-Reconstruction Upscaler, not a Super-Res network. Just in case some people get too excited about this being suggested.

**Comment by Xaymar** [comment]
I will decline this as an addition to Video Super-Resolution, as it does not fit the definition of the filter, but will not decline it in its entirety. This may be available as a simple Video Upscaling filter in StreamFX, or perhaps OBS Studio itself adds it to the list of scaling methods available.

**Comment by Xaymar** [comment]
With v0.11, the `Video Super-Resolution` filter has been renamed to `Upscaling`. This allows (lesser) upscaling algorithms to share the same feature name, but they will not look as high quality as a proper upscaling method.
